### PR TITLE
docs (documentation-website): Contributing.md and /contributing page …

### LINKED ISSUE #524: The Contributing guides don't match
## Context

Right now the Contributing.md and the /contributing page on the website have small differences.

## Goal

Fix the differences in a sensible manner, so that both are a valid source of the same information,

### PR DESCRIPTION
**Pull request by martincarapia**
…are consistent with each other

# The Pull Request is ready

- [x ] fixes #524 
- [x ] all actions are passing
- [ x] only fixes a single issue

## Overview

Spelling mistakes fixed. Information is consistent with one another. 

## Documentation

- [ x] all documentation is in correct english
- [x ] the documentation explains used words, that are not common
- [x ] there are no abbreviations used

Documentation-website en.yml also changed. Added more list_item(s) to match CONTRIBUTING.md

## Summary by CodeRabbit

- **Documentation**
    - Enhanced contributing guidelines for improved readability and clarity.
    - Reordered subprojects and adjusted contribution workflow in project documentation.
    - Updated mindmap section, testing requirements, and added setup instructions.
    - Improved commit message guidelines and added new list items to the contributing section.

### CODE REVIEW

File: documentation-website/language/en.yml [hunk 1]
```diff
@@ -205,8 +206,8 @@ contributing:
       list_item_4: "kebab-case for file names"
     requirements:
       title: "Requirements"
-      node: "One of the last two node versions"
-      os: "OS: Linux / Windows / Mac"
+      node: "One of the last two node versions is required."
+      os: "OS: Linux / Windows *(Other OS were not tested)*"
```
Comment (Idrinth) [inline documentation-website/language/en.yml:210]: the original is right here actually

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -227,7 +226,7 @@ behaviour in order to illustrate the impact of the change.
 ##### Commit Message Footer
 
 The footer can contain information about breaking changes and deprecation
-and is also the place to reference GitHub issues and other PR's that this
+and is also the place to reference GitHub issues and other PRs that this
```
Comment (Idrinth) [inline CONTRIBUTING.md:229]: please expand the PR to Pull Request

File: documentation-website/language/en.yml [hunk 3]
```diff
@@ -205,8 +206,8 @@ contributing:
       list_item_4: "kebab-case for file names"
     requirements:
       title: "Requirements"
-      node: "One of the last two node versions"
-      os: "OS: Linux / Windows / Mac"
+      node: "One of the last two node versions is required."
+      os:  "Supported Operating Systems: Linux, Windows, and macOS."
```
Comment (Idrinth) [inline documentation-website/language/en.yml:210]: please add macOS as a new line to .dictionary.txt

**Review by Idrinth — CHANGES_REQUESTED** [review]

**Review by Idrinth — CHANGES_REQUESTED** [review]

**Comment by Idrinth** [comment]
@martincarapia did a lot of performance optimizations, sorry about adding conflicts :/

File: .dictionary.txt [hunk 4]
```diff
@@ -47,3 +47,5 @@ gitea
 Gitea
 history-microservice
 api-bench-cli
+cli
```
Comment (Idrinth) [inline .dictionary.txt:50]: CLI is upper case only to my knowledge

**Review by Idrinth — APPROVED** [review]

File: .dictionary.txt (hunk 4 above)
Comment (Idrinth) [reply .dictionary.txt:50]: yes please

File: documentation-website/language/en.yml [hunk 5]
```diff
@@ -183,25 +183,24 @@ contributing:
...
       list_item_1: "fork the repository"
-      list_item_2: "create a feature branch (issue-xxx-does-something) from master"
-      list_item_3: "create a draft pull request"
-      list_item_4: "mention fixes #xxx in the description"
-      list_item_5: "make your changes to the branch"
-      list_item_6: "check if actions are all green"
-      list_item_7: "mark as ready to review"
-      list_item_8: "get changes reviewed"
-      list_item_9: "if review is positive get changes merged otherwise fix and send back to review"
-    translation:
```
Comment (Idrinth) [inline documentation-website/language/en.yml:198]: this is missing

**Review by Idrinth — CHANGES_REQUESTED** [review]

**Comment by martincarapia** [comment]
Sorry about that. I'm guessing I accidentally deleted it while fixing conflicts.

**Review by Idrinth — DISMISSED** [review]

**Pull request merged**